src/github/entities/user-installation.entity.ts

```diff
@@ -0,0 +1,15 @@
+import { Base } from 'src/universal/base.entity';
+import { Column, Entity, ManyToOne } from 'typeorm';
+import { User } from './user.entity';
+
+@Entity()
+export class UserInstallation extends Base {
+  @Column({ type: 'varchar', length: 20 })
+  installationId: string;
+
+  @Column({ type: 'varchar', length: 255 })
+  organizationName: string;
+
+  @ManyToOne(() => User, (user) => user.installations)
+  user: User;
```

Comment:
PullProofAI

There is a potential performance issue in the `UserInstallation` entity related to the `@ManyToOne` relationship with the `User` entity. When loading `UserInstallation` entities, it will also eagerly load associated `User` entities by default, which can lead to performance degradation, especially if there are many `User` entities.

#### Recommendation
Consider using lazy loading for the `User` relationship in the `UserInstallation` entity to avoid unnecessary eager loading of `User` entities when querying `UserInstallation` instances.

```typescript
@ManyToOne(() => User, (user) => user.installations, { lazy: true })
user: Promise<User>;
```

By making this change, the associated `User` entities will only be loaded when accessed, improving performance by reducing unnecessary data retrieval.


<!-- This is an auto-generated comment by PullProofAI -->